orttraining/orttraining/python/training/ortmodule.py

```diff
@@ -372,27 +437,30 @@ def _get_forward_graph(self, input_names, dynamic_axes, *inputs, **kwargs):
         TODO: How to support dynamic axes? Dimensions are determined by samples
         TODO: How to ingest **kwargs in proper order during export?
         '''
-        # Export the model to memory
-        f = io.BytesIO()
 
-        # Deepcopy inputs, since input values may change after model run.
-        # NOTE: Inputs may contain tensors that have attributes preventing their deepcopy (example grad_fn).
-        # Therefore, deepcopy only the data component of the input tensors for export.
-        sample_inputs_copy = []
-        for model_input in inputs:
-            sample_inputs_copy.append(model_input.data if isinstance(model_input, torch.Tensor) else model_input)
-        sample_inputs_copy = copy.deepcopy(tuple(sample_inputs_copy))
+        # Setup dynamic axes for onnx model
```

Comment:
We no longer need to pass in the ```input_names``` and ```dynamic_axes``` variable to this function. Also, since we are parsing the inputs for onnx export here, we can do away with parsing it in the function that invokes ```_get_forward_graph``` (```_get_forward_graph_and_init_gradient_graph_builder```).
```self._input_names_require_grad``` can be set inside this function.

Reply:
Same as above. rebase failed